src/app/auth/auth.service.ts

```diff
@@ -1,29 +1,88 @@
 import { Injectable } from '@angular/core';
-import { HttpClient, HttpHeaders, HttpErrorResponse } from '@angular/common/http';
+import { HttpClient, HttpHeaders, HttpErrorResponse, HttpResponse } from '@angular/common/http';
 import { Observable, throwError, catchError, of, BehaviorSubject } from 'rxjs';
-import { Router } from '@angular/router';
+import { map } from 'rxjs';
+import { LoginResponse } from '../shared/models/loginResponse'
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthService {
-  private tokenName = "token";
-  private authStateSubject = new BehaviorSubject<boolean>(this.hasToken());
-  authStatus: Observable<boolean> = this.authStateSubject.asObservable();
+  private _tokenName = "token";
+  private _apiUrl = 'https://rb-messenger.azurewebsites.net';
 
-  constructor() {}
+  private _authStateSubject = new BehaviorSubject<boolean>(this.hasToken());
+  authStatus$: Observable<boolean> = this._authStateSubject.asObservable();
+
+  constructor(private _http: HttpClient) {}
+
+  login(username: string, password: string): Observable<HttpResponse<LoginResponse>> {
```

Comment:
```suggestion
  login(username: string, password: string): Observable<LoginResponse> {
```
You don't need the HttpResponse here, just return Observable of your Interface